### PR TITLE
PF-1201: Generate sensible resrouce id when it is not specified for Controlled resources

### PR DESCRIPTION
**Pull request by yuhuyoyo**
Currently, user needs to specify both resource name and the resource-type specific name (e.g. datasetId, bucket name, etc).

Instead, we can just generate a default name if it's not specified.
To create a BQ dataset, one can say
terra resource create bq-dataset --name=foo
WSM will then create a bq-dataset of resource name foo and datasetId foo given "foo" is a valid dataset name as well. If the resource name is not a valid datasetid, we will generate a unique dataset id terra_uuid_dataset.

For create a GCS bucket, we will create a bucket of name terra_uuid_bucket

For AiNotebook, we will also reuse the resource name if it is a valid ai notebook name. Otherwise, we will generate a unique ai notebook name.

### CODE REVIEW

**Review by ddietterich — APPROVED** [review]
LGTM - a style suggestion

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledAiNotebookInstanceResource.java [hunk 1]
```diff
@@ -146,6 +146,10 @@ public int hashCode() {
     return result;
   }
 
+  public static String generateUniqueInstanceId() {
+    return String.format("terra-%s-notebook", UUID.randomUUID());
```
Comment (zloery) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledAiNotebookInstanceResource.java:150]: Why do notebooks need random IDs?

Agree this is necessary for buckets (which are globally namespaced), but my understanding is that you could just use the resource name here.

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java [hunk 2]
```diff
@@ -345,7 +344,10 @@ private ApiCloneControlledGcpGcsBucketResult fetchCloneGcsBucketResult(
             .accessScope(AccessScopeType.fromApi(body.getCommon().getAccessScope()))
             .managedBy(managedBy)
             .applicationId(controlledResourceService.getAssociatedApp(managedBy, userRequest))
-            .datasetName(body.getDataset().getDatasetId())
+            .datasetName(
+                body.getDataset().getDatasetId() == null
```
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java:348]: nice!

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledAiNotebookInstanceResource.java (hunk 1 above)
Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledAiNotebookInstanceResource.java:150]: ah, you are right. i also realize that the resource name is not always a valid instance id. so I made some change.

**Review by zloery — APPROVED** [review]

**Pull request merged**